Add actions workflow for checklist submission

kaitj opened the pull request:
This PR is part of a bigger group of PRs across the proceedings and checklist repos to add functionality for simplifying / streamlining the checklist submission process:

1. Added a couple of issue templates for (a) submission of a checklist (with `checklist` label) and (b) "other" issue template, intended for all other issues such as bugs, features, etc. The checklist template is also used to pre-fill checklist from the submissions page.

2. Added an actions CI, which triggers on the addition of the `approved` label (together with the `checklist` label) will:
    a. Extract the checklist data from the issue
    b. Create the necessary file and validate it against the schema (grabbed from the checklist repo)
    c. Push to the proceedings page before closing the issue if no errors

kaitj committed:
Add issue templates

- Add issue template for checklist submission
- Add a generic issue template for reporting a bug / feature request

kaitj committed:
Add checklist CI

- Adds a GH action that creates and pushes the new checklist to the
  proceedings page when an issue is tagged with "approved"
- Add validation check of checklists against schema (schema
  taken from the standards-checklist

shnizzedy commented:
Do we have any documentation (in addition this PR and https://github.com/nmind/standards-checklist/pull/43) about the design and flow of using issues for staging / approving submitted checklists? I think it makes sense, but I don't want to leave it implicit

kaitj commented:
That's a good point and we probably do want to have it explicitly documented. I haven't across documentation yet, is there a file or repo that exists for documentation - happy to write this up.

shnizzedy commented:
> is there a file or repo that exists for documentation

I'm not sure! I didn't remember this plan (if I ever knew it) but figured it out from the code changes (I think ― it's like 

```mermaid
flowchart LR
webapp[user submitted checklist from web app] --> issue[nmind/proceedings issue] --"NMIND moderator approves checklist issue"--> proceedings[user submission added to https://www.nmind.org/proceedings/]
```

right?)

Maybe just in the README here might make sense?

It would probably be good for users to know what the process is too, though, so maybe somewhere on nmind.org that https://www.nmind.org/standards-checklist/ and https://www.nmind.org/proceedings/ can link to?

kaitj commented:
That makes sense and yep that would be the process. I’ll add it to the readme to start and try to see where it might fit on the website too.

Commit:
Add checklist workflow to README

Commit:
Add process + agreement to checklist issue template

- Added the mermaid flowchart illustrating the checklist submission
  process
- Added a checkbox to the checklist as a required field such that the
  user has one additional check prior to agreeing to submitting their
checklist.

kaitj commented:
Added identical workflow info to the README both here and in the `standards-checklist` repo (will update PR there so it is noted).

I looked around to see where adding the information about the workflow might best go on the public facing pages, but hadn't found anything I felt fit in nicely. For now I've added a reference to the relevant README section in the issue template. Aside from adding an image and linking that, I don't believe that the issue template markdown is able to handle mermaid graphs (or at least I couldn't get it to work) that would make this process more prominent. I've also added a required checkbox for submitters "agreeing to the submission process", which will (hopefully) at a minimum avoid blind clickthrough submissions.

shnizzedy commented:
> Added identical workflow info […] I looked around to see where adding the information about the workflow might best go on the public facing pages, but hadn't found anything I felt fit in nicely.

I think this is definitely enough for now, but eventually we'll want it on the site somewhere and a [single source of truth](https://en.wikipedia.org/wiki/Single_source_of_truth)
